src/engine.rs

```diff
@@ -20,14 +24,10 @@ pub fn run(function_path: PathBuf, input_path: PathBuf) -> Result<FunctionRunRes
     let module = Module::from_file(&engine, &function_path)
         .map_err(|e| anyhow!("Couldn't load the Function {:?}: {}", &function_path, e))?;
 
-    let input: serde_json::Value = serde_json::from_reader(
-        std::fs::File::open(&input_path)
-            .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?,
-    )
-    .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?;
-    let input = serde_json::to_vec(&input)?;
+    let mut buffer = Vec::new();
+    input.read_to_end(&mut buffer)?;
```

Comment:
Any reason why you are reading all of stdin into a buffer just to then create a `Cursor` again?

Reply:
This moved a bit so we could validate the input as JSON before we attempt execution.

I ran into issues with `AsRef<[u8]>` being a non-auto trait. I think we could a trait that implements all supertraits of Read, Sync, `static, and AsRef<[u8]>.

Since we want to validate that the input is valid JSON I have it reading it in so the same `Vec<u8>` can be used once for validation and wrapping with the Cursor.